internal/pkg/service/stream/storage/statistics/repository/stats_allocation.go

```diff
@@ -10,13 +10,20 @@ import (
 
 	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop/iterator"
 	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop/op"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/plugin"
 	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/level"
 	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/model"
 	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/statistics"
 )
 
 func (r *Repository) estimateSliceSizeOnSliceOpen() {
 	r.plugins.Collection().OnSliceOpen(func(ctx context.Context, now time.Time, file model.File, slice *model.Slice) error {
+		// Optional: If Sink is new, we can skip calculation, there is no previous slice/statistics
+		if sink := plugin.SinkFromContext(ctx, file.SinkKey); sink != nil && sink.CreatedAt().Time().Equal(now) {
+			slice.LocalStorage.AllocatedDiskSpace = file.LocalStorage.DiskAllocation.ForNextSlice(0)
```

Comment:
Added optional condition, callculation of pre-allocated space for new slices has no effect, if the sink is new, so there is no previous slice/stats. This removes some weird/unexpected query. 

See added snapshots.